asserters.go

```diff
@@ -0,0 +1,54 @@
+package safecast
+
+import "fmt"
+
+func assertNotNegative[T Type](i T) error {
+	if i < 0 {
+		return fmt.Errorf("%w: %v is negative", ErrIntegerOverflow, i)
+	}
+	return nil
+}
+
+func checkUpperBoundary[T Type](value T, boundary uint64) error {
+	if value <= 0 {
+		return nil
+	}
+
+	var bigger bool
+	switch f := any(value).(type) {
+	case float32:
+		bigger = float64(f) >= float64(boundary)
+	case float64:
+		bigger = float64(f) >= float64(boundary)
```

Comment:
same code? did you mean float32 above? add a test that fails before fixing?

Reply:
No it's the right way to do.

I'll add a comment to explain

Reply:
> No it's the right way to do.

Unless I'm wrong 😅

I'll check

Reply:
No, it's normal.

I added comments, it should be clearer